### PR TITLE
mgr/telemetry: add mempool stats to telemetry perf report

### PR DESCRIPTION
**Pull request by ljflores**
Adding mempool stats to the telemetry perf report for a deeper look at osd memory usage.

Example of the `mempool` entry in the perf report:

```
"mempool": {
        "osd.0": {
            "bloom_filter": {
                "bytes": 0,
                "items": 0
            },
            "bluefs": {
                "bytes": 5280,
                "items": 104
            },
            "bluefs_file_reader": {
                "bytes": 12864,
                "items": 6
            },
            "bluefs_file_writer": {
                "bytes": 744,
                "items": 3
            },
            "bluestore_Blob": {
                "bytes": 23816,
                "items": 229
            },
            "bluestore_Buffer": {
                "bytes": 6432,
                "items": 67
            },
            "bluestore_Extent": {
                "bytes": 10992,
                "items": 229
            },
            "bluestore_SharedBlob": {
                "bytes": 25648,
                "items": 229
            },
            "bluestore_alloc": {
                "bytes": 1120,
                "items": 14
            },
            "bluestore_cache_data": {
                "bytes": 306796,
                "items": 73
            },
            "bluestore_cache_meta": {
                "bytes": 236166,
                "items": 36088
            },
            "bluestore_cache_onode": {
                "bytes": 437240,
                "items": 643
            },
            "bluestore_cache_other": {
                "bytes": 39692,
                "items": 1151
            },
            "bluestore_fsck": {
                "bytes": 0,
                "items": 0
            },
            "bluestore_inline_bl": {
                "bytes": 138234,
                "items": 643
            },
            "bluestore_txc": {
                "bytes": 800,
                "items": 1
            },
            "bluestore_writing": {
                "bytes": 364850,
                "items": 85
            },
            "bluestore_writing_deferred": {
                "bytes": 16000,
                "items": 4
            },
            "buffer_anon": {
                "bytes": 2295438,
                "items": 80
            },
            "buffer_meta": {
                "bytes": 43296,
                "items": 492
            },
            "mds_co": {
                "bytes": 0,
                "items": 0
            },
            "osd": {
                "bytes": 3196544,
                "items": 272
            },
            "osd_mapbl": {
                "bytes": 0,
                "items": 0
            },
            "osd_pglog": {
                "bytes": 570440,
                "items": 1097
            },
            "osdmap": {
                "bytes": 332600,
                "items": 1535
            },
            "osdmap_mapping": {
                "bytes": 0,
                "items": 0
            },
            "pgmap": {
                "bytes": 0,
                "items": 0
            },
            "unittest_1": {
                "bytes": 0,
                "items": 0
            },
            "unittest_2": {
                "bytes": 0,
                "items": 0
            }
        },
```

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ljflores** [comment]
@markhpc would you suggest keeping the OSDs separate here, or combining them for an average? Or perhaps providing both a "snapshot" of one OSD + an average?

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -894,6 +930,7 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
             report['stat_sum_per_pool'] = self.get_stat_sum_per_pool()
             report['io_rate'] = self.get_io_rate()
             report['osd_perf_histograms'] = self.get_osd_histograms()
+            report['mempool'] = self.get_mempool()
```
[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/telemetry/module.py:933]: Oh I see, thanks

**Comment by ljflores** [comment]
I'm leaving the commits separated now, but let me know if I should squash them. Or if anything else needs to be changed.

**Review by yaarith — APPROVED** [review]

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/yuriw-2021-10-31_23:32:39-rados-wip-yuri2-testing-2021-10-31-0642-distro-basic-smithi
http://pulpito.front.sepia.ceph.com/yuriw-2021-10-31_15:22:06-rados-wip-yuri2-testing-2021-10-31-0642-distro-basic-smithi/

Failures unrelated, tracked in:

https://tracker.ceph.com/issues/37657
https://tracker.ceph.com/issues/50280

Details:

Bug #37657: Command failed on smithi075 with status 1: 'sudo yum install -y kernel' - Ceph - Ceph
Bug #50280: cephadm: RuntimeError: uid/gid not found - Ceph - Ceph

**Comment by yuriw** [comment]
ref: https://trello.com/c/En6EisxC

**Comment by ljflores** [comment]
@neha-ojha @yaarith I'm not approving this one since it's my own PR, but I believe it can be merged.

**Pull request merged**